src/datastore/postgres-store.ts

```diff
@@ -1607,6 +1608,129 @@ export class PgDataStore extends (EventEmitter as { new (): DataStoreEventEmitte
     return { results: parsed, total: count };
   }
 
+  async searchHash({
+    hash,
+  }: {
+    hash: string;
+  }): Promise<{ found: false } | { found: true; result: DbSearchResult }> {
+    const txQuery = await this.pool.query<{ tx_id: Buffer }>(
+      `SELECT tx_id FROM txs WHERE tx_id = $1 LIMIT 1`,
+      [hexToBuffer(hash)]
+    );
+    if (txQuery.rowCount > 0) {
+      return {
+        found: true,
+        result: { entity_type: 'tx_id', entity_id: bufferToHexPrefixString(txQuery.rows[0].tx_id) },
+      };
+    }
+
+    const txMempoolQuery = await this.pool.query<{ tx_id: Buffer }>(
+      `SELECT tx_id FROM mempool_txs WHERE tx_id = $1 LIMIT 1`,
+      [hexToBuffer(hash)]
+    );
+    if (txMempoolQuery.rowCount > 0) {
+      return {
+        found: true,
+        result: {
+          entity_type: 'mempool_tx_id',
+          entity_id: bufferToHexPrefixString(txMempoolQuery.rows[0].tx_id),
+        },
+      };
+    }
+
+    const blockQueryResult = await this.pool.query<{ block_hash: Buffer }>(
+      `SELECT block_hash FROM blocks WHERE block_hash = $1 LIMIT 1`,
+      [hexToBuffer(hash)]
+    );
+    if (blockQueryResult.rowCount > 0) {
+      return {
+        found: true,
+        result: {
+          entity_type: 'block_hash',
+          entity_id: bufferToHexPrefixString(blockQueryResult.rows[0].block_hash),
+        },
+      };
+    }
+    return { found: false };
+  }
+
+  async searchPrincipal({
+    principal,
+  }: {
+    principal: string;
+  }): Promise<{ found: false } | { found: true; result: DbSearchResult }> {
+    const isContract = principal.includes('.');
+    const entityType = isContract ? 'contract_address' : 'standard_address';
+    const successResult: DbSearchResult = {
```

Comment:
Maybe just have this variable be `{ found: true, result: { entityType, entityId }`. Then you could just `return successResult` instead of the expanded format for each query.